### PR TITLE
Public IP support and SAML Auth docs

### PR DESCRIPTION
**Pull request by aaha97**

### CODE REVIEW

File: docs/crds/avinfrasetting.md [hunk 1]
```diff
@@ -122,6 +122,13 @@ AviInfraSetting CRD can be used to enable/disable Route Health Injection (RHI) o
 
 This overrides the global `enableRHI` flag for the virtualservices corresponding to the AviInfraSetting.
 
+#### Enable/Disable Public IP
+
+AviInfraSetting CRD can be used to enable/disable Public IP on the virtualservices created by AKO. 
+
+        network:
+          enablePublicIP: true
```
Comment (apalsule) [inline docs/crds/avinfrasetting.md:130]: can you also add a statement that this is supported only on public clouds.

Comment (aaha97) [reply docs/crds/avinfrasetting.md:130]: updated as a note

**Review by chauhanshubham — APPROVED** [review]

**Comment by chauhanshubham** [comment]
lgtm, could you please add one liners for these in the CHANGELOG too @aaha97

File: docs/crds/avinfrasetting.md [hunk 2]
```diff
@@ -122,6 +122,15 @@ AviInfraSetting CRD can be used to enable/disable Route Health Injection (RHI) o
...
+
+Note that enabling Public IP is only supported for **public clouds**.
```
Comment (sudswasavi) [inline docs/crds/avinfrasetting.md:132]: is it supported for all public clouds or just AWS?

File: docs/values.md [hunk 3]
```diff
@@ -205,6 +205,12 @@ The `cidr` within `vipNetworkList` is not required for AWS and Azure Cloud.
 The username/password of the Avi controller is specified with this flag. The username/password are base64 encoded by helm and a corresponding secret
 object is used to maintain the same. Editing this field requires a restart (delete/re-create) of the AKO pod.
 
+### avicredentials.authtoken
+
+The generated authtoken from the Avi controller can be specified with this flag as an alternative to password. The authtoken is also base64 encoded
+and maintained by secret object. The token refresh is managed by AKO. In case of token refresh failure, a new token needs to be generated and updated
```
Comment (sudswasavi) [inline docs/values.md:211]: The authtoken is also base64 encoded and updated regularly in a Kubernetes Secret object.

**Review by sudswasavi — COMMENTED** [review]
Couple of nits.

File: docs/crds/avinfrasetting.md (hunk 2 above)
Comment (chauhanshubham) [reply docs/crds/avinfrasetting.md:132]: It is for all public clouds.

**Review by chauhanshubham — APPROVED** [review]

File: docs/values.md (hunk 3 above)
Comment (aaha97) [reply docs/values.md]: updated

File: docs/crds/avinfrasetting.md (hunk 2 above)
Comment (aaha97) [reply docs/crds/avinfrasetting.md:132]: yes, for all public clouds

**Review by monotosh-avi — APPROVED** [review]

**Pull request merged**